### PR TITLE
Refactoring of filters

### PR DESCRIPTION
**Pull request by marcuslinke**
This PR eleminates the use of the generic `Filters` object in the command APIs. Instead there are now seperate methods for each command specific filter.

### CODE REVIEW

**Comment by marcuslinke** [comment]
After merging https://github.com/docker-java/docker-java/tree/issue-356 into master refactoring filters for list networks command is needed also.

File: src/test/java/com/github/dockerjava/netty/exec/ListImagesCmdExecTest.java [hunk 1]
```diff
@@ -95,8 +95,6 @@ private String createDanglingImage() {
         String imageId = dockerClient.commitCmd(container.getId()).exec();
 
         dockerClient.stopContainerCmd(container.getId()).exec();
-        dockerClient.killContainerCmd(container.getId()).exec();
         dockerClient.removeContainerCmd(container.getId()).exec();
         return imageId;
-    }
-}
+    }}
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/netty/exec/ListImagesCmdExecTest.java]: undo?

**Comment by marcuslinke** [comment]
Closing this in favor of merged PR https://github.com/docker-java/docker-java/pull/426

**Comment by KostyaSha** [comment]
No need to merge master, you can just `git rebase master` branch locally, then review and force push back.

**Comment by KostyaSha** [comment]
Plus it will keep authors.

**Pull request closed without merging**